File: backend/Ledgerly.Application/DTOs/Bill/BillDto.cs
namespace Ledgerly.Application.DTOs.Bill;

public class BillDto
{
    public Guid Id { get; set; }
    public Guid OrganizationId { get; set; }
    public Guid CustomerId { get; set; }
    public string CustomerName { get; set; } = string.Empty;
    public string? BillNumber { get; set; }
    public decimal TotalAmount { get; set; }
    public decimal OutstandingAmount { get; set; }
    public DateTime BillDate { get; set; }
    public string Status { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; }
    public List<BillItemDto> Items { get; set; } = new();
}

File: app/lib/models/bill/bill_dto.dart
import 'package:json_annotation/json_annotation.dart';
import 'bill_item_dto.dart';

part 'bill_dto.g.dart';

@JsonSerializable()
class BillDto {
  final String id;
  final String organizationId;
  final String customerId;
  final String customerName;
  final String? billNumber;
  final double totalAmount;
  final double outstandingAmount;
  @JsonKey(name: 'billDate')
  final DateTime billDate;
  final String status;
  @JsonKey(name: 'createdAt')
  final DateTime createdAt;
  final List<BillItemDto> items;

  BillDto({
    required this.id,
    required this.organizationId,
    required this.customerId,
    required this.customerName,
    this.billNumber,
    required this.totalAmount,
    required this.outstandingAmount,
    required this.billDate,
    required this.status,
    required this.createdAt,
    required this.items,
  });

  factory BillDto.fromJson(Map<String, dynamic> json) =>
      _$BillDtoFromJson(json);

  Map<String, dynamic> toJson() => _$BillDtoToJson(this);
}

